tests/ops/functions/test_equal.py

```diff
@@ -1772,6 +1772,22 @@ def test_sum_with_multi_wire_operations(self, base_list1, base_list2, res):
         op2 = qml.sum(*base_list2)
         assert qml.equal(op1, op2) == res
 
+    def test_sum_equal_order_invarient(self):
+        """Test that the order of operations doesn't affect equality"""
+        H1 = qml.prod(qml.PauliX(0), qml.PauliX(1))
+        H2 = qml.s_prod(1.0, qml.sum(qml.PauliY(0), qml.PauliY(1)))
+
+        true_res = qml.sum(
+            qml.s_prod(2j, qml.prod(qml.PauliZ(0), qml.PauliX(1))),
+            qml.s_prod(2j, qml.prod(qml.PauliX(0), qml.PauliZ(1))),
+        )
+        true_res = true_res.simplify()
```

Comment:
Do you understand why this example breaks when removing the `true_res.simplify()` here?
Both results are `Sum` instances now and seem equivalent, with only the order of appearance differeing, which I would have thought is fixed by the updated `_sort_key` method in `Sum` 🤔 

```python
H1 = qml.prod(qml.PauliX(0), qml.PauliX(1))
H2 = qml.s_prod(1., qml.sum(qml.PauliY(0), qml.PauliY(1)))
true_res = qml.sum(qml.s_prod(2j, qml.prod(qml.PauliZ(0), qml.PauliX(1))), qml.s_prod(2j, qml.prod(qml.PauliX(0), qml.PauliZ(1))))
res = qml.prod(H1, H2) - qml.prod(H2, H1)
res = res.simplify()

>>> type(true_res), type(res)
(pennylane.ops.op_math.sum.Sum, pennylane.ops.op_math.sum.Sum)
>>> true_res, res
((2j*(PauliZ(wires=[0]) @ PauliX(wires=[1]))) + (2j*(PauliX(wires=[0]) @ PauliZ(wires=[1]))),
 (2j*(PauliX(wires=[1]) @ PauliZ(wires=[0]))) + (2j*(PauliZ(wires=[1]) @ PauliX(wires=[0]))))
```

Reply:
I'm thinking that maybe `equal` for `Sum` should just use the hash, because (as [we note in its definition](https://github.com/PennyLaneAI/pennylane/blob/f854611abc354f480a3b04894f0956fb4d3d3b8e/pennylane/ops/op_math/sum.py#L163-L166)) we shouldn't care about operand order in Sums, and the hash doesn't. I confirmed locally (on master..) that `res` and `true_res` have the same hash here.

`Prod.hash` does care about sorting, so we should be careful here.

Reply:
I was thinking of something similar originally but learned that hashes should not be used for equality comparison because you can just happen to have the same hash even though the operators are different (I dont know how un/likely that is though). Would certainly be easier and faster though..

Reply:
Found that on the one true source of wisdom: stackexchange/overflow but I can't find the thread now. If you say this is safe/good(enough) practice I'd be happy if we took this on 👍

Reply:
Addressed, see my comment below.